### PR TITLE
feat: Option to choose the number of results on the Catalog page in the configuration menu.

### PR DESCRIPTION
**Pull request by Dryrtan**
My idea initially was to do the same with the search page, but my knowledge in Electro is limited and prevented me from doing this, but it is open to anyone who wants to implement it on the search page.

### CODE REVIEW

**Comment by hydralauncher** [comment]
Could you attach a screenshot?

**Comment by Dryrtan** [comment]
> Thanks for the contribution! Could you please attach a screenshot?

Of course

resultsPerPage

**Comment by zamitto** [comment]
These radio button seems more like check boxes. I think they should have a round style that radio buttons usually have

**Comment by Dryrtan** [comment]
> These radio button seems more like check boxes. I think they should have a round style that radio buttons usually have

Alright, I'll make the change

**Comment by Dryrtan** [comment]
> These radio button seems more like check boxes. I think they should have a round style that radio buttons usually have

The new style of the radio component

image

File: src/renderer/src/pages/catalogue/catalogue.tsx [hunk 1]
```diff
@@ -34,13 +34,34 @@ export function Catalogue() {
     navigate(`/game/${game.shop}/${game.objectID}`);
   };
 
+  const [paginationSettings, setPaginationSettings] = useState({ resultsPerPage: 30 });
+
+  useEffect(() => {
+    async function fetchUserPreferences() {
+      try {
+        const [, userPreferences] = await Promise.all([
+          window.electron.getDefaultDownloadsPath(),
```
Comment (zamitto) [inline src/renderer/src/pages/catalogue/catalogue.tsx:43]: do you need to call getDefaultDownloadsPath? you dont seem to be using its result

File: src/renderer/src/pages/catalogue/catalogue.tsx [hunk 2]
```diff
@@ -34,13 +34,34 @@ export function Catalogue() {
     navigate(`/game/${game.shop}/${game.objectID}`);
   };
 
+  const [paginationSettings, setPaginationSettings] = useState({ resultsPerPage: 30 });
+
+  useEffect(() => {
+    async function fetchUserPreferences() {
+      try {
```
Comment (zamitto) [inline src/renderer/src/pages/catalogue/catalogue.tsx:41]: i'm not sure if this try catch block is needed

**Comment by Dryrtan** [comment]
Alright, refactoring...

**Comment by hydralauncher** [comment]
Closing this in wishes for a more intuitive approach to handle this.

**Pull request closed without merging**